src/HealthChecks.AzureServiceBus/AzureServiceBusSubscriptionHealthCheck.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.ServiceBus.Management;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+
+namespace HealthChecks.AzureServiceBus
+{
+    public class AzureServiceBusSubscriptionHealthCheck : IHealthCheck
+    {
+        private static readonly ConcurrentDictionary<string, ManagementClient> ManagementClientConnections =
+            new ConcurrentDictionary<string, ManagementClient>();
+        private readonly string _connectionString;
+        private readonly string _topicName;
+        private readonly string _subscriptionName;
+
+
+        public AzureServiceBusSubscriptionHealthCheck(string connectionString, string topicName, string subscriptionName)
+        {
+            if (string.IsNullOrEmpty(connectionString))
+            {
+                throw new ArgumentNullException(nameof(connectionString));
+            }
+
+            if (string.IsNullOrEmpty(topicName))
+            {
+                throw new ArgumentNullException(nameof(topicName));
+            }
+
+            if (string.IsNullOrEmpty(subscriptionName))
+            {
+                throw new ArgumentNullException(nameof(subscriptionName));
+            }
+
+            _connectionString = connectionString;
+            _topicName = topicName;
+            _subscriptionName = subscriptionName;
+        }
+
+        public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = default)
+        {
+            try
+            {
+                var connectionKey = $"{_connectionString}_{_topicName}_{_subscriptionName}";
+                if (!ManagementClientConnections.TryGetValue(connectionKey, out var managementClient))
+                {
+
```

Comment:
```suggestion
```

Reply:
hmm, why?

Reply:
There is no semantic load in an empty line in this place.

Reply:
Will remove it